model.go

```diff
@@ -6,18 +6,31 @@ import (
 	"os"
 )
 
-type Param struct {
+type Model []Layer
+
+type Layer struct {
 	Bias         []float64       `json:"bias"`         //バイアス
 	KW           int             `json:"kW"`           //フィルタの幅
 	KH           int             `json:"kH"`           //フィルタの高さ
-	Weight       [][][][]float64 `json:"weight"`       //重み
 	NInputPlane  int             `json:"nInputPlane"`  //入力平面数
 	NOutputPlane int             `json:"nOutputPlane"` //出力平面数
+	Weight       [][][][]float64 `json:"weight"`       //重み
+	WeightVec    []float64
 }
 
-type Model []Param
+func flattenWeight(weight [][][][]float64) []float64 {
+	var vec []float64
+	for _, v3 := range weight {
+		for _, v2 := range v3 {
+			for _, v1 := range v2 {
+				vec = append(vec, v1...)
+			}
+		}
+	}
+	return vec
+}
 
-func LoadModelFile(path string) (*Model, error) {
+func LoadModelFile(path string) (Model, error) {
```

Comment:
__[golint]__
* [golint] exported function LoadModelFile should have comment or be unexported